chain-core/src/state/account/op/data/attribute.rs

```diff
@@ -0,0 +1,42 @@
+use parity_scale_codec::{Decode, Encode, Error, Input, Output};
+#[cfg(not(feature = "mesalock_sgx"))]
+use serde::{Deserialize, Serialize};
+
+/// attributes in StakedState-related transactions
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+#[cfg_attr(not(feature = "mesalock_sgx"), derive(Serialize, Deserialize))]
+pub struct StakedStateOpAttributes {
+    pub chain_hex_id: u8,
+    pub app_version: u64,
+}
+
+impl StakedStateOpAttributes {
+    pub fn new(chain_hex_id: u8) -> Self {
+        StakedStateOpAttributes {
+            chain_hex_id,
+            app_version: crate::APP_VERSION,
+        }
+    }
+}
+
+impl Encode for StakedStateOpAttributes {
+    fn encode_to<EncOut: Output>(&self, dest: &mut EncOut) {
+        dest.push_byte(self.chain_hex_id);
+        dest.push(&self.app_version);
+    }
+
+    fn size_hint(&self) -> usize {
+        5
```

Comment:
```suggestion
        self.chain_hex_id.size_hint() + self.app_version.size_hint()
```
Seems all the hardcoded size hint treat u64 as 4 bytes, I think it should be 8 bytes? Replace the hardcoded numbers with actual calculations?

Reply:
true, I'd look where it's possible